bin/reth/src/node/mod.rs

```diff
@@ -295,7 +295,9 @@ impl<Ext: RethCliExt> NodeCommand<Ext> {
         let head = self.lookup_head(Arc::clone(&db)).wrap_err("the head block is missing")?;
 
         // setup the blockchain provider
-        let factory = ProviderFactory::new(Arc::clone(&db), Arc::clone(&self.chain));
+        let (highest_snapshots_tx, highest_snapshots_rx) = watch::channel(None);
+        let factory = ProviderFactory::new(Arc::clone(&db), Arc::clone(&self.chain))
```

Comment:
am i correct to understand that this will be shared across all components that might create providers?

Reply:
through `blockchain_db`

Reply:
yes